executables/Bench.hs

```diff
@@ -2,15 +2,15 @@
 
 module Main where
 
-#if !defined(ghcjs_HOST_OS)
+#if !defined(ghcjs_HOST_OS) && defined(BUILD_TOOL)
 import qualified Pact.Bench as Bench
 #endif
 
 
 main :: IO ()
 main =
-#if defined(ghcjs_HOST_OS)
-    error "Error: command line REPL does not exist in GHCJS mode"
-#else
+#if !defined(ghcjs_HOST_OS) && defined(BUILD_TOOL)
     Bench.main
+#else
+    error "Error: command line REPL does not exist in GHCJS mode"
```

Comment:
I wonder if the CPP in this module can be removed all together because building the bench component is disabled in the cabal file if `build-tool` is set to `False`.

```cabal
if !flag(build-tool)
  buildable: False
```

So, it should not be able to trigger this error. It would only be possible if someone would erroneously add this module to the `lib:pact` component in the cabal file.

Reply:
The following would keep the CPP more contained and would fail already at compile time:

```haskell
#if !defined(ghcjs_HOST_OS)
#error "Error: command line REPL does not exist in GHCJS mode"
#elif defined(BUILD_TOOL)
#error "Error: command line REPL is not support when the build-tool flag is disabled."
#endif

main :: IO ()
main = Bench.main
```

To the best of my knowledge the `#error` directive is part of the CPP standard.

Reply:
I'm trying out the no-CPP approach in #888